# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/AnnexB.h
# Commit: 2d92dd455f19
# Full Hash: 2d92dd455f198471c7e02d315ed241a289737a10
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-05 21:26:57
# Regressor Bug: 1852333
# File Overlap Count: 4
# Description:
#   Bug 1852333 - part2 : detect config change for HEVC stream. r=media-playback-reviewers,padenot
#   
#   This patch detects inband change for the HEVC playback so that we can
#   recreate decoder if the new config is different from the previous one.
#   
# ==============================================================================

diff -r ee2c23cc0a50 -r 2d92dd455f19 dom/media/platforms/agnostic/bytestreams/AnnexB.h
--- a/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Thu Oct 05 04:59:44 2023 +0000
+++ b/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Thu Oct 05 04:59:44 2023 +0000
@@ -43,6 +43,8 @@
   static bool ConvertSampleToAVCC(
       mozilla::MediaRawData* aSample,
       const RefPtr<mozilla::MediaByteBuffer>& aAVCCHeader = nullptr);
+  // Convert a sample from Annex B to HVCC. An HVCC extradata must not be set.
+  static bool ConvertSampleToHVCC(mozilla::MediaRawData* aSample);
 
   // Covert sample to 4 bytes NALU byte stream.
   static mozilla::Result<mozilla::Ok, nsresult> ConvertAVCCTo4BytesAVCC(